packages/client/src/internals/dataLoader.ts

```diff
@@ -18,6 +20,15 @@ type BatchLoader<TKey, TValue> = {
   };
 };
 
+/**
+ * A function that should never be called unless we messed something up.
+ */
+const throwFatalError = () => {
+  throw new Error(
+    'Something went wrong. Please submit an issue at https://github.com/trpc/trpc/issues/new',
+  );
+};
```

Comment:
Using this fn in places allowed me to skip a bunch of typecasting when creating objects